Src/library/base/elks/kernel/tuple.e

```diff
@@ -369,22 +443,65 @@ feature -- Element change
 			valid_index: valid_index (index)
 			valid_type_for_index: valid_type_for_index (v, index)
 		do
-			inspect eif_item_type ($Current, index)
-			when boolean_code then eif_put_boolean_item_with_object ($Current, index, $v)
-			when character_8_code then eif_put_character_8_item_with_object ($Current, index, $v)
-			when character_32_code then eif_put_character_32_item_with_object ($Current, index, $v)
-			when real_64_code then eif_put_real_64_item_with_object ($Current, index, $v)
-			when real_32_code then eif_put_real_32_item_with_object ($Current, index, $v)
-			when pointer_code then eif_put_pointer_item_with_object ($Current, index, $v)
-			when natural_8_code then eif_put_natural_8_item_with_object ($Current, index, $v)
-			when natural_16_code then eif_put_natural_16_item_with_object ($Current, index, $v)
-			when natural_32_code then eif_put_natural_32_item_with_object ($Current, index, $v)
-			when natural_64_code then eif_put_natural_64_item_with_object ($Current, index, $v)
-			when integer_8_code then eif_put_integer_8_item_with_object ($Current, index, $v)
-			when integer_16_code then eif_put_integer_16_item_with_object ($Current, index, $v)
-			when integer_32_code then eif_put_integer_32_item_with_object ($Current, index, $v)
-			when integer_64_code then eif_put_integer_64_item_with_object ($Current, index, $v)
-			when Reference_code then eif_put_reference_item_with_object ($Current, index, $v)
+			inspect item_code (index)
+			when boolean_code then
+				check valid_type_for_index: attached {BOOLEAN_REF} v as l_b then
```

Comment:
I thought more about it, and I have the impression that your goal is to use 'if' instead of 'check' unless the compiler complains about it. In this particular case, I agree that the compiler will not complain and that the code will do what is expected. But the use of 'if' looks like defensive programming to me: why test something in the body of the routine which is already expressed in the precondition. Semantically, 'check' is different: in theory if the compiler can prove that the assertion is true at compile-time, then the test can be bypassed at run-time.
Now if the goal is to have zero 'check' in the code, that's another story....